### PR TITLE
6

### PR DESCRIPTION
**Pull request by schoeyy**
/api/articles/:article_id/comments

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -119,4 +136,44 @@ describe("Articles", () => {
...
+        .get("/api/articles/invalidId/comments")
+        .expect(400)
+        .then((response) => {
+          const { msg } = response.body;
+          expect(msg).toBe("Bad Request: This is not a valid article number!");
+        });
+    });
+    test("Status: 404 || article_id not found", () => {
+      return request(app)
+        .get("/api/articles/10000/comments")
```
Comment (leekli) [inline __tests__/app.test.js:171]: Good check for valid article id which does not exist

File: controllers/nc-news.controller.js [hunk 2]
```diff
@@ -36,6 +37,17 @@ exports.getArticleById = (req, res, next) => {
     });
 };
 
+exports.getArticleCommentsById = (req, res, next) => {
+  const { article_id } = req.params;
+  fetchArticleCommentsById(article_id)
+    .then((comments) => {
+      res.status(200).send({ comments });
```
Comment (leekli) [inline controllers/nc-news.controller.js:44]: Yes, good use of an appropriately named key, 'comments' which represents multiple comments - good work!

File: controllers/nc-news.controller.js [hunk 3]
```diff
@@ -36,6 +37,17 @@ exports.getArticleById = (req, res, next) => {
...
+    })
+    .catch((err) => {
+      next(err);
+    });
```
Comment (leekli) [inline controllers/nc-news.controller.js:48]: 👍

File: models/nc-news.model.js [hunk 4]
```diff
@@ -21,9 +21,32 @@ exports.fetchArticleById = (article_id) => {
...
+  `,
+      [article_id]
+    )
+    .then((result) => {
+      if (result.rowCount == 0) {
+        return Promise.reject({
+          code: 404,
+          msg: `Not Found: Article ${article_id} cannot be found!`,
+        });
+      } else {
```
Comment (leekli) [inline models/nc-news.model.js:40]: So.... is this doing what you want?

Is it:
* article exists but has no comments? or.....
* article does not exist? ....

Have another look at this. This is something where you need to know in advance if the article id exists. We had a look at this on Wednesday and here are the notes: https://notes.northcoders.com/courses/js-back-end/advanced-error-handling

You'll need to implement functionality to check an article id exists, and if it doesn't... reject it, if it does... then go get the comments.

**Review by leekli — CHANGES_REQUESTED** [review]
Hey Tom,

Well done so far! There are some changes needed, please see my comments.

Essentially:
* test is missing an author property
* a .length check in the test
* you're going need an 'check article exists' function to check an article exists first, before going to do a comments query. NC notes if needed: https://notes.northcoders.com/courses/js-back-end/advanced-error-handling

File: __tests__/app.test.js [hunk 5]
```diff
@@ -143,15 +143,25 @@ describe("Articles", () => {
         .expect(200)
         .then((response) => {
           const { comments } = response.body;
+          expect(comments.length > 0).toBe(true);
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:146]: I think you can be more specific here. What length do you expect this array to be?

File: models/nc-news.model.js [hunk 6]
```diff
@@ -10,11 +10,11 @@ exports.fetchArticleById = (article_id) => {
   return db
     .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
     .then((result) => {
-      if (result.rowCount == 0) {
+      if (result.rows.length == 0) {
```
Comment (Beth-McCormack) [inline models/nc-news.model.js:13]: This isnt quite implementing Lee's previous comment. You should create a util function which checks whether the article_id exists or not, then either rejects or gets the comments for said article 🙂

**Pull request merged**